submissions/akulahere/task-js-post-oop/index.js

```diff
@@ -0,0 +1,76 @@
+class Inhabitant {
+  constructor(species, name, gender, legs, saying) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.saying = saying;
+    this.friends = 'No friends' ;
+  }
+
+  makeFriend(friend) {
+    this.friends = this.friends === 'No friends' ? [] : this.friends;
+    this.friends.push(friend.name);
+  }
+  getProperties() {
+    return ['species', 'name', 'gender', 'legs', 'saying', 'friends'];
+  }
+
+  toString() {
+    return this.getProperties().map(property => `${capitalizeString(property)}: ${this[property]}`).join(' ')
+  }
+}
+
+class Dog extends Inhabitant {
+  constructor(name, gender) {
+    super('dog', name, gender, 4, 'woof-woof!');
+  }
+}
+
+class Cat extends Inhabitant {
+  constructor(name, gender) {
+    super('cat', name, gender, 4, 'mrrr!');
+  }
+}
+
+class CatWoman extends Cat {
+  constructor(name) {
+    super(name, 'female');
+    this.species = 'CatWoman';
+    this.legs = 2;
```

Comment:
If we need to access inherited props directly then this is a clear sign that something is not really OK in the design.
In fact the compositional pattern is better than OOP when "hybrids" are required.
You did your best within OOP paradigm.
This was an intentional pitfall to learn that OOP has its own deficiencies.